### PR TITLE
Correctly convert Python float to float64 when passing argument as Tensor

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #136413
* #136599

I can't actually test the Dynamo codegen fix as it is impossible to
directly use the Tensor at the moment.

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

### CODE REVIEW

**Review by bobrenjc93 — APPROVED** [review]

File: torch/_dynamo/codegen.py [hunk 1]
```diff
@@ -469,7 +469,7 @@ def make_call_generated_code(self, fn_name: str) -> None:
                     lambda: self.extend_output(
                         [
                             self.create_load_python_module(torch),
-                            self.create_load_attr("as_tensor"),
+                            self.create_load_attr("_as_tensor_fullprec"),
```
Comment (int3) [inline torch/_dynamo/codegen.py:472]: This breaks `TORCH_LOGS` due to a circular import:

```
Traceback (most recent call last):
  File "/data/users/jezng/pytorch/benchmarks/dynamo/torchbench.py", line 13, in <module>
    import torch
  File "/data/users/jezng/pytorch/torch/__init__.py", line 2619, in <module>
    _logging._init_logs()
  File "/data/users/jezng/pytorch/torch/_logging/_internal.py", line 928, in _init_logs
    _update_log_state_from_env()
  File "/data/users/jezng/pytorch/torch/_logging/_internal.py", line 738, in _update_log_state_from_env
    log_state = _parse_log_settings(log_setting)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/users/jezng/pytorch/torch/_logging/_internal.py", line 714, in _parse_log_settings
    elif _is_valid_module(name):
         ^^^^^^^^^^^^^^^^^^^^^^
  File "/data/users/jezng/pytorch/torch/_logging/_internal.py", line 728, in _is_valid_module
    __import__(qname)
  File "/data/users/jezng/pytorch/torch/_inductor/autotune_process.py", line 28, in <module>
    import torch._inductor.async_compile  # noqa: F401 required to warm up AsyncCompile pools
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/users/jezng/pytorch/torch/_inductor/async_compile.py", line 16, in <module>
    from torch._dynamo.device_interface import get_registered_device_interfaces
  File "/data/users/jezng/pytorch/torch/_dynamo/__init__.py", line 3, in <module>
    from . import convert_frame, eval_frame, resume_execution
  File "/data/users/jezng/pytorch/torch/_dynamo/convert_frame.py", line 53, in <module>
    from . import config, exc, trace_rules
  File "/data/users/jezng/pytorch/torch/_dynamo/trace_rules.py", line 46, in <module>
    from .variables import (
  File "/data/users/jezng/pytorch/torch/_dynamo/variables/__init__.py", line 2, in <module>
    from .builtin import BuiltinVariable
  File "/data/users/jezng/pytorch/torch/_dynamo/variables/builtin.py", line 27, in <module>
    from ..guards import GuardBuilder, install_guard
  File "/data/users/jezng/pytorch/torch/_dynamo/guards.py", line 327, in <module>
    "___as_tensor": torch._as_tensor_fullprec,
                    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/data/users/jezng/pytorch/torch/__init__.py", line 2563, in __getattr__
    raise AttributeError(f"module '{__name__}' has no attribute '{name}'")
```

Comment (int3) [reply torch/_dynamo/codegen.py:472]: actually no, I bisected it to https://github.com/pytorch/pytorch/pull/136365

Comment (ezyang) [reply torch/_dynamo/codegen.py:472]: does https://github.com/pytorch/pytorch/pull/136548 fix yours

Comment (int3) [reply torch/_dynamo/codegen.py:472]: Yep, it fixes things. Thanks!

**Comment by ezyang** [comment]
Waiting on https://github.com/pytorch/pytorch/pull/136548 to land

**Pull request closed without merging**